### ISSUE TITLE
The window session popup only shows the first two windows

### ISSUE THREAD

**Issue by hesselmonk** [fabiospampinato/chrome-window-session#1]
When saving more than 2 sessions the list, that pops up when clicking the Window Session icon in the extensions bar, only contains the first two saved windows.

The badge that shows the amount of tabs in the window session does update, but the window itself does not show up in the list.

Clicking it in a window that has not been saved yet does show the input field to name the window. But naming it doesn't cause it show up in the list.

(Windows 10, Chrome 63.0.3239.132 (Official Build) (64-bit) (cohort: Stable))

**Comment by fabiospampinato** [comment]
Thanks for reporting the bug! 

Unfortunately I can't reproduce this, I need to do it in order to fix it.

**Comment by hesselmonk** [comment]
I know _very little_ about chrome extensions, but it seems that upon inspecting the extensions' background after a fresh start of the browser, it starts with two errors:

> Unchecked runtime.lastError while running browserAction.setBadgeText: No tab with id: 241.
    at Object.update (chrome-extension://koeeegbghbokaeaikifjhdbbkjbolcep/background.js:17352:30)
background.html:1 Unchecked runtime.lastError while running browserAction.setBadgeBackgroundColor: > No tab with id: 241.
    at Object.update (chrome-extension://koeeegbghbokaeaikifjhdbbkjbolcep/background.js:17353:30)

The two lines it refers to (I assume they were transpiled, so line numbers might be useless) are :
```
        chrome.browserAction.setBadgeText({ text, tabId });
        chrome.browserAction.setBadgeBackgroundColor({ color: config_1.default.badge.color, tabId });
 ```

The error gets re-thrown every time I close a tab, and every time the tab id is different.

**Comment by hesselmonk** [comment]
I just noticed that closing a window also throws an error:
> Unchecked runtime.lastError while running windows.get: No window with id: 687.
    at resolve (chrome-extension://koeeegbghbokaeaikifjhdbbkjbolcep/background.js:17224:28)
    at Object.get (chrome-extension://koeeegbghbokaeaikifjhdbbkjbolcep/background.js:17223:16)
    at Object.update (chrome-extension://koeeegbghbokaeaikifjhdbbkjbolcep/background.js:17311:37)
    at Object.onChanged (chrome-extension://koeeegbghbokaeaikifjhdbbkjbolcep/background.js:17784:32)

It seems to happen in this code:
```
    /* HELPERS */
    async get(id) {
        return new Promise(resolve => {
            chrome.windows.get(id, { populate: true }, resolve);
        });
    },
```

**Comment by hesselmonk** [comment]
Ok, more info:

When trying to add a session (after the initial two) the following error gets thrown after pressing return in the popup's input field:

> Unchecked runtime.lastError while running storage.set: QUOTA_BYTES_PER_ITEM quota exceeded
    at resolve (chrome-extension://koeeegbghbokaeaikifjhdbbkjbolcep/background.js:17720:33)
    at Object.set (chrome-extension://koeeegbghbokaeaikifjhdbbkjbolcep/background.js:17719:16)
    at Object.map (chrome-extension://koeeegbghbokaeaikifjhdbbkjbolcep/background.js:17734:26)

Interestingly, the extension icon _does_ get a badge with the (correct) tabs count, it won't appear in the popup though. After restarting the browser the count is missing again, as if the session wasn't stored at all.

**Comment by fabiospampinato** [comment]
Thank you for your messages.

From the last stack trace it seems the maximum storage quota is reached for some reason, I'll have to investigate further on this.

I'm not sure when this will be fixed given that I'm currently pretty busy with other stuff, in case you'd like to submit a PR I'll be happy to review it and hopefully merge it.

**Comment by hesselmonk** [comment]
I wish I could, but I have 0 working knowledge about Babel and practically no chrome plugin development experience either :/

If I look at the data in chrome.storage.sync though it's name2window key is pretty big and the window2name key contains the windows I wanted to save, but aren't appearing in the json in name2window.
![image](https://user-images.githubusercontent.com/28805623/36379967-de3b7050-1580-11e8-847b-88fc24f63ecf.png)

Adding a new window with just one tab in it ("www.google.com") works fine, so I assume the stored name2window just exceed some maximum size limit once I add a new window with actual tabs in it? Without knowing anything about the actual implementation, could it be an idea to store a key for each window and use the window2name lookup to find that key directly instead of in the name2window data?

**Comment by hesselmonk** [comment]
Did you find time to look into this issue?

**Comment by fabiospampinato** [comment]
@hesselmonk not yet, unfortunately. I've been keeping this tab open on Chrome as a reminder though. If you have the time it would be great if you could try to find the problem and submit a pull request for it, I can't say when I'll have to time for fixing this.

**Comment by fabiospampinato** [comment]
I addressed every problem I could find in v1.1.0, which is pending review, now the extension should actually work. Check out the commit history to learn what changed.
